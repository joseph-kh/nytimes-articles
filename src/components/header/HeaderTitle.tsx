import React from "react";
import { StyleSheet, View } from "react-native";
import SemiBoldText from "@/components/ui/texts/SemiBoldText";
import { horizontalScale, moderateScale } from "@/utils/scaler";
import COLORS from "@/theme/colors";

type Props = {
  children: React.ReactNode;
};

const HeaderTitle: React.FC<Props> = (props) => {
  const { children } = props;

  return (
    <View style={styles.header}>
      <SemiBoldText
        style={styles.headerText}
        numberOfLines={1}
        ellipsizeMode="tail"
      >
        {children}
      </SemiBoldText>
    </View>
  );
};

const styles = StyleSheet.create({
  header: {
    paddingRight: horizontalScale(20),
    width: moderateScale(200),
  },

  headerText: {
    fontSize: moderateScale(20),
    color: COLORS.white,
  },
});

export default HeaderTitle;
