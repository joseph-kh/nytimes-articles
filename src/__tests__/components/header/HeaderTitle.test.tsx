import React from "react";
import renderer from "react-test-renderer";
import HeaderTitle from "@/components/header/HeaderTitle";

describe("HeaderTitle Component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<HeaderTitle>Header Text</HeaderTitle>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
