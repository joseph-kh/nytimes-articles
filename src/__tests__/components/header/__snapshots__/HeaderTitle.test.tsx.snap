// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderTitle Component renders correctly 1`] = `
<View
  style={
    {
      "paddingRight": 40,
      "width": 300,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "SemiBold",
        "fontSize": 30,
      }
    }
  >
    Header Text
  </Text>
</View>
`;
